packages/block-library/src/block/edit.js

```diff
@@ -45,8 +55,38 @@ export default function ReusableBlockEdit( { attributes: { ref } } ) {
 		ref
 	);
 
+	const alignments = [ 'wide', 'full' ];
+
+	useEffect( () => {
```

Comment:
I think the presence of the `setAttributes` calls in this `useEffect` results in a post's Update button not being disabled when reloaded in the editor, since the editor believes that something has changed. Here's a quick gif of reloading the editor:

![2023-09-06 14 38 25](https://github.com/WordPress/gutenberg/assets/14988353/5d312b63-6a45-4265-8255-a6589e741837)

Reply:
yeh, I don't like setAttribs in useEffects, but was struggling to see a different way to handle this - but will take another look.

Reply:
trying to handle this outside a useEffect causes trying to render while something else is still rendering errors. I have added __unstableMarkNextChangeAsNotPersistent for now which prevents the editor from seeing the post as dirty after the updates. Is sort of nice having this anyway as makes it clear that these attribute changes are not persisted.

Let me now if you can see a way of running this on changes to `blocks` without using a `useEffect`.

Reply:
Can we move this logic into the reusable block creation action instead ?

Reply:
> Can we move this logic into the reusable block creation action instead ?

The source pattern can be edited via the `wp_block` post type editor as well. We'd need to make sure the pattern instance reflected any changes in child alignment there. Same for the frontend, once we add the wrapper.

There's a more detailed comment on this in the PR description for https://github.com/WordPress/gutenberg/pull/54289 which attempts to add the frontend wrapper in a BC way.